video/out/opengl/lcms.c

```diff
@@ -242,23 +313,12 @@ bool gl_lcms_get_lut3d(struct gl_lcms *p, struct lut3d **result_lut3d)
     if (!profile)
         goto error_exit;
 
-    // We always generate the 3DLUT against BT.2020, and transform into this
-    // space inside the shader if the source differs.
-    struct mp_csp_primaries csp = mp_get_csp_primaries(MP_CSP_PRIM_BT_2020);
-
-    cmsCIExyY wp = {csp.white.x, csp.white.y, 1.0};
-    cmsCIExyYTRIPLE prim = {
-        .Red   = {csp.red.x,   csp.red.y,   1.0},
-        .Green = {csp.green.x, csp.green.y, 1.0},
-        .Blue  = {csp.blue.x,  csp.blue.y,  1.0},
-    };
+    cmsHPROFILE vid_profile = get_vid_profile(cms, profile, prim, trc);
+    if (!vid_profile) {
+        cmsCloseProfile(profile);
+        goto error_exit;
+    }
 
-    // 2.4 is arbitrarily used as a gamma compression factor for the 3DLUT,
-    // reducing artifacts due to rounding errors on wide gamut profiles
-    cmsToneCurve *tonecurve = cmsBuildGamma(cms, 2.4);
-    cmsHPROFILE vid_profile = cmsCreateRGBProfileTHR(cms, &wp, &prim,
-                        (cmsToneCurve*[3]){tonecurve, tonecurve, tonecurve});
-    cmsFreeToneCurve(tonecurve);
     cmsHTRANSFORM trafo = cmsCreateTransformTHR(cms, vid_profile, TYPE_RGB_16,
                                                 profile, TYPE_RGB_16,
                                                 p->opts.intent,
```

Comment:
That should be a separate patch (as mentioned). How would you want this to be user-facing, some flag like `icc-apple-compat` or something?.

That could actually be used for the above, where BT.1886 would decode as 1.961 in `icc-apple-compat` mode.